sunkit_image/coalignment/interface.py

```diff
@@ -0,0 +1,155 @@
+import warnings
+from typing import NamedTuple
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+from sunpy.util.exceptions import SunpyUserWarning
+from sunpy.sun.models import differential_rotation
+
+from sunkit_image.coalignment.decorators import registered_methods
+
+__all__ = ["coalign", "AffineParams"]
+
+
+class AffineParams(NamedTuple):
+    """
+    A 2-element tuple containing scale values defining the image scaling along the x and y axes.
+    """
+    scale: tuple[float, float]
+    """
+    A 2x2 matrix defining the rotation transformation of the image.
+    """
+    rotation_matrix: tuple[tuple[float, float], tuple[float, float]]
+    """
+    A 2-element tuple stores the translation of the image along the x and y axes.
+    """
+    translation: tuple[float, float]
+
+
+def update_fits_wcs_metadata(reference_map, target_map, AffineParams):
+    """
+    Update the metadata of a sunpy Map object based on affine transformation
+    parameters.
+
+    Parameters
+    ----------
+    reference_map : `sunpy.map.Map`
+        The reference map object to which the target map is to be coaligned.
+    target_map : `sunpy.map.Map`
+        The original map object whose metadata is to be updated.
+    AffineParams : object
+        An object containing the affine transformation parameters. This object must
+        have attributes for translation (dx, dy), scale, and rotation.
+
+    Returns
+    -------
+    `sunpy.map.Map`
+        A new sunpy map object with updated metadata reflecting the affine transformation.
+    """
+    # Extacting the affine parameters
+    pc_matrix = target_map.rotation_matrix
+    translation = AffineParams.translation
+    scale = AffineParams.scale
+    rotation_matrix = AffineParams.rotation_matrix
+    # Updating the PC matrix
+    new_pc_matrix = pc_matrix @ rotation_matrix
+    # Calculate the new reference pixel. 
+    old_reference_pixel = np.array([target_map.reference_pixel.x.value, target_map.reference_pixel.y.value])
+    new_reference_pixel = scale*rotation_matrix @ old_reference_pixel + translation
+    reference_coord = reference_map.wcs.pixel_to_world(new_reference_pixel[0],new_reference_pixel[1])
+    Txshift = reference_coord.Tx - target_map.reference_coordinate.Tx
+    Tyshift = reference_coord.Ty - target_map.reference_coordinate.Ty
+    # Create a new map with the updated metadata
+    fixed_map = target_map.shift_reference_coord(Txshift, Tyshift)
+
+    fixed_map.meta["PC1_1"] = new_pc_matrix[0, 0]
+    fixed_map.meta["PC1_2"] = new_pc_matrix[0, 1]
+    fixed_map.meta["PC2_1"] = new_pc_matrix[1, 0]
+    fixed_map.meta["PC2_2"] = new_pc_matrix[1, 1]
+
+    fixed_map.meta['cdelt1'] = (target_map.scale[0] / scale[0]).value
+    fixed_map.meta['cdelt2'] = (target_map.scale[1] / scale[1]).value
+
+    return fixed_map
+
+
+def warn_user_of_separation(reference_map,target_map):
```

Comment:
I think the consensus is to make this a private method since it is only used once internally.